source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,237 @@
+===========
+Aggregation
+===========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Aggregation operations process data in your MongoDB collections and return computed results. The **aggregation pipeline**
+is like a factory assembly line. Along the line are assembly stations, called **aggregation stages** . Like an assembly
+station configured with specific tooling to perform its task, each ``stage`` uses **expressions** to perform some
+operation or combination of operations.
+
+Aggregation Does More
+~~~~~~~~~~~~~~~~~~~~~
+
+Using ``find`` operations, you can:
+
+- select *what* documents to return
+- select *what* fields to return
+- sort the results
+
+Using ``aggregation`` operations, you can:
+
+- perform all ``find`` operations
+- rename fields
+- calculate fields
+- summarize data
+- group values
+
+Illustrated Example
+~~~~~~~~~~~~~~~~~~~
+
+In a car factory assembly line, one station might shape and bend the frame using a hydraulic press and a welder, another
+might affix the doors with a drill, and so on.
+
+In the following animation, every document in the collection enters the pipeline and is first filtered by
+the ``$match`` stage. In this example, the stage selectively matches documents that are either "red" or "blue",
+filtering "green" documents out. The documents continue flowing through the pipeline and enter the ``$project`` stage,
+which transforms them from squares to circles. Lastly, the documents flow into the ``$group`` stage, configured to
+produce the ratio of red to blue circles.
+
+.. figure:: /figures/agg_factory.gif
+   :border:
+   :alt: Animation showing how the aggregation pipeline works, conceptually.
+
+
+Using the **Aggregation builders** API, the same operation may look something like:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/agg-factory.java
+   :language: java
+
+.. code-block:: json
+
+   {"_id": 0, "red": 3, "blue": 2}
+
+Terms
+~~~~~
+
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+- :ref:`Aggregation Builders <builders>`
+
+Runnable Examples
+-----------------
+
+The following examples require a ``restaurants`` collection in the ``test`` database. To create and populate the collection,
+follow the `directions <https://github.com/mongodb/docs-assets/tree/drivers>`_ to download and load the dataset. If you
+prefer to load the data directly and have both ``mongoimport`` and ``curl`` in your path:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Atlas
+
+      .. code-block:: sh
+
+         mongoimport --uri mongodb+srv://<your-atlas-uri>/test --collection restaurants --drop --file <(curl https://raw.githubusercontent.com/mongodb/docs-assets/drivers/restaurants.json)
```

Comment:
Nitpick: what happens here when curl fails? Maybe we can instead split it into two commands so that the user would see the related error from curl before attempting mongoimport? Same comment for localhost tab.

Reply:
I'm not convinced the value separating the two would provide in the unlikely scenario the curl fails for a small 10 document demo collection is worth worrying over. Open to more discussion on this in private channels.